src/xcode/ENA/ENA/Source/Scenes/Settings/SettingsViewController.swift

```diff
@@ -134,18 +134,20 @@ final class SettingsViewController: UITableViewController {
 	private func notificationSettings() {
 		let currentCenter = UNUserNotificationCenter.current()
 
-		currentCenter.requestAuthorization(options: [.alert, .sound, .badge]) { [weak self] granted, error in
+		currentCenter.getNotificationSettings { [weak self] (settings) in
 			guard let self = self else { return }
 
-			if let error = error {
-				log(message: "Error while requesting notifications permissions: \(error.localizedDescription)")
+			switch settings.authorizationStatus {
```

Comment:
Switch is not needed here IMO. There are only two states when notifications are enabled, ```.authorized``` and ```.provisional```.

```swift
if (settings.authorizationStatus == .authorized || settings.authorizationStatus == .provisional) && (self.store.allowRiskChangesNotification || self.store.allowTestsStatusNotification) {
      self.settingsViewModel.notifications.setState(state: true)
} else {
      self.settingsViewModel.notifications.setState(state: false)
}
```

Something like that. That's also encouraged by Apple in the [official doc](https://developer.apple.com/documentation/usernotifications/asking_permission_to_use_notifications).